### ISSUE TITLE
ATS A11y POC

### ISSUE THREAD

**Issue by ravenokko** [shopware/platform#8735]
### User story

We need to adapt our test suite to protect not only functional requirements from regressions, but also accessibility features. Our methods must balance performance with the impact on our coverage from the lens of legal compliance. Ideally, we might uncover more preventative based measures (ala linting) against both common accessibility failures and the ones that risk being overlooked. 

### Acceptance criteria

- Adapt key ATS workflows to use only keyboard functionality for interactions
- Adapt key ATS workflows to use screen reader functionality with GuidePup
- Add tests for A11y specific features and functionality
- Write documentation to ensure proper manual testing is covered (and carried out) as part of our Definition of Done
